packages/test/src/test-worker-deployment-versioning.ts

```diff
@@ -0,0 +1,409 @@
+/**
+ * Tests worker-deployment-based versioning
+ *
+ * @module
+ */
+import assert from 'assert';
+import { randomUUID } from 'crypto';
+import asyncRetry from 'async-retry';
+import { ExecutionContext } from 'ava';
+import { Client } from '@temporalio/client';
+import { toCanonicalString, WorkerDeploymentVersion } from '@temporalio/common';
+import { temporal } from '@temporalio/proto';
+import { Worker } from './helpers';
+import { Context, makeTestFunction } from './helpers-integration';
+import { unblockSignal, versionQuery } from './workflows';
+
+const test = makeTestFunction({ workflowsPath: __filename });
+
+test('Worker deployment based versioning', async (t) => {
+  const taskQueue = 'worker-deployment-based-versioning-' + randomUUID();
```

Comment:
nit: Next time, consider using the `const { startWorkflow, createWorker, client, taskQueue } = helpers(t)` function. That can save quite a few lines of boilerplate.

Reply:
I couldn't use those helpers for reasons I can't remember perfectly, now - I think it had to do with the fact that I need multiple workers and need them all to use the same TQ but different other options.